packages/utils/src/PromiseUtils.ts

```diff
@@ -1,32 +1,42 @@
 import CanceledPromiseError from './CanceledPromiseError';
 import TimeoutError from './TimeoutError';
 
+export interface CancelablePromise<T> extends Promise<T> {
+  cancel: () => void;
+}
+
 class PromiseUtils {
-  static isCancelablePromise(promise) {
+  static isCancelablePromise(
+    promise: unknown
+  ): promise is CancelablePromise<unknown> {
+    const cancelablePromise = promise as CancelablePromise<unknown>;
     return (
       this.isPromise(promise) &&
-      promise.cancel != null &&
-      typeof promise.cancel === 'function'
+      cancelablePromise.cancel != null &&
+      typeof cancelablePromise.cancel === 'function'
     );
   }
 
-  static isPromise(promise) {
+  static isPromise(promise: unknown): promise is Promise<unknown> {
     return (
       promise != null &&
-      promise.then != null &&
-      typeof promise.then === 'function'
+      (promise as Promise<unknown>).then != null &&
+      typeof (promise as Promise<unknown>).then === 'function'
```

Comment:
Talked w/ Bender about these and we came to the conclusion we can just remove these 2 type guards. They're not actually used anywhere except unit tests. So remove these 2 methods + corresponding unit tests